### PR TITLE
[Test][NSA] Mark flaky GQA window sliding config as xfail

### LINKED ISSUE #346: [TEST][NSA] fix flaky test_nsa_gqa_window_sliding_op causing 45% CI failures
### Symptom / Motivation
`test_nsa_gqa_window_sliding_op` is a flaky test that accounts for approximately 45% of CI failures. When it fails, it produces `max_err=6.16` against `atol=3e-3`, which is roughly 2000x over the tolerance threshold.

The specific failing configuration is:
```
[1-16-1024-1024-64-128-True-32--1-dtype0-accum_dtype0-False]
```

This has been observed failing across multiple CI runs:
- Run 22699721206 (main push)
- Run 22668049247
- Run 22670725809
- Run 22698382826

### Root Cause Analysis
The NSA GQA window sliding kernel produces numerically divergent results for this specific parameter combination. The error magnitude (2000x over tolerance) suggests this is not a precision edge case but a genuine correctness bug in the kernel for this configuration. The root cause in the kernel itself needs separate investigation.

### Related Files
- `testing/test_nsa_attention.py` — the flaky test
- NSA kernel implementation (GQA window sliding path)

## Goal
Prevent this flaky test from blocking CI while the underlying kernel correctness issue is investigated and fixed separately.

## Plan

1. Mark `test_nsa_gqa_window_sliding_op` with `@pytest.mark.xfail` for the failing configuration to unblock CI
2. Investigate the kernel correctness issue for the specific parameter combination as a follow-up

## Constraints
- Do not remove or skip the test entirely — use `@pytest.mark.xfail` so the test still runs and will alert when fixed
- The kernel correctness fix should be tracked as a separate issue

## Acceptance Criteria
- [ ] Modified files pass unit tests
- [ ] The flaky configuration is marked with `@pytest.mark.xfail`
- [ ] CI pass rate is no longer impacted by this test
- [ ] The test automatically starts passing again (xfail strict) once the kernel is fixed

### PR DESCRIPTION
**Pull request by lcy-seso**
Closes #346

## Summary

- Mark the flaky `test_nsa_gqa_window_sliding_op` configuration `(1-16-1024-1024-64-128-True-32--1-dtype0-accum_dtype0-False)` with `@pytest.mark.xfail(strict=True)` to unblock CI
- The test still runs but is expected to fail; `strict=True` ensures it will alert (XPASS) when the kernel bug is fixed
- The underlying kernel correctness issue (max_err=6.16 vs atol=3e-3) should be investigated separately

## Test plan

- [x] pre-commit passed
- [x] `python -m pytest --collect-only` confirms all 3 test cases collected with xfail marker on the flaky config
- [x] Syntax validation passed

### CODE REVIEW

File: tests/ops/test_deepseek_nsa_gqa_window_sliding.py [hunk 1]
```diff
@@ -13,7 +13,14 @@ class GqaWindowSlidingFixture(FixtureBase):
...
          "window_size_right, dtype, accum_dtype, tune", [
-             (1, 16, 1024, 1024, 64, 128, True, 32, -1, torch.float16, torch.float32, False),
+             pytest.param(
+                 1, 16, 1024, 1024, 64, 128, True, 32, -1, torch.float16, torch.float32, False,
+                 marks=pytest.mark.xfail(
+                     strict=True,
+                     reason="NSA GQA window sliding kernel correctness bug: "
+                            "max_err=6.16 vs atol=3e-3 (see issue #346)",
+                 ),
+             ),
```
[the remark this answers is not in the document]
Comment (lcy-seso) [reply tests/ops/test_deepseek_nsa_gqa_window_sliding.py:25]: Accepted. Added `id="gqa-window-sliding-flaky-bug-346"` to the pytest.param. See 88bf97d.

File: tests/ops/test_deepseek_nsa_gqa_window_sliding.py [hunk 2]
```diff
@@ -13,7 +13,14 @@ class GqaWindowSlidingFixture(FixtureBase):
     PARAMS = [
         ("batch_size, groups, uq, ukv, heads, dim, is_causal, window_size_left, "
          "window_size_right, dtype, accum_dtype, tune", [
-             (1, 16, 1024, 1024, 64, 128, True, 32, -1, torch.float16, torch.float32, False),
+             pytest.param(
+                 1, 16, 1024, 1024, 64, 128, True, 32, -1, torch.float16, torch.float32, False,
+                 marks=pytest.mark.xfail(
+                     strict=True,
```
[the remark this answers is not in the document]
Comment (lcy-seso) [reply tests/ops/test_deepseek_nsa_gqa_window_sliding.py]: Declined. The error is 2000x over tolerance (max_err=6.16 vs atol=3e-3), which indicates a genuine kernel correctness bug rather than flakiness from precision edge cases. `strict=True` is required by the issue constraints so we're alerted when the kernel is fixed.

[the remark this answers is not in the document]
Comment (lcy-seso) [reply tests/ops/test_deepseek_nsa_gqa_window_sliding.py]: Accepted. Added `raises=AssertionError` to scope the xfail to the known assertion failure only. See 4598eb3.

[the remark this answers is not in the document]
Comment (lcy-seso) [reply tests/ops/test_deepseek_nsa_gqa_window_sliding.py]: Revisited: CI confirmed the test is hardware-dependent (XPASS on CI runner). Changed to `strict=False` as suggested. See 9ef57f2.
